### PR TITLE
Kakarot node setup

### PR DESCRIPTION
**Pull request by tcoratger**
Related https://github.com/paradigmxyz/reth/blob/main/crates/ethereum/node/src/node.rs

### CODE REVIEW

File: crates/primitives/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn it_works() {
+        let result = add(2, 2);
+        assert_eq!(result, 4);
+    }
+}
```
Comment (greged93) [inline crates/primitives/src/lib.rs:17]: should we remove this?

File: crates/pool/src/validate/mod.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+use reth_transaction_pool::{
+    validate::{EthTransactionValidator, EthTransactionValidatorBuilder},
+    BlobStore,
+};
+use std::sync::Arc;
+
+/// A wrapper around the Reth [`EthTransactionValidatorBuilder`].
+/// The produced Validator will reject EIP4844 transactions not supported by Kakarot at the moment.
+#[derive(Debug)]
+pub struct KakarotTransactionValidatorBuilder(EthTransactionValidatorBuilder);
```
Comment (greged93) [inline crates/pool/src/validate/mod.rs:13]: Is this used anywhere? I think it should be passed somewhere at some point right?

Comment (tcoratger) [reply crates/pool/src/validate/mod.rs:13]: Oups this is true, just forgot to rearrange. I have just created a Kakarot Transaction Validator which is obtained from this builder and which is then used in the general Validator.

Comment (greged93) [reply crates/pool/src/validate/mod.rs:13]: Did you push? I can't find the Kakarot Transaction Validator

File: crates/pool/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+[lints]
+workspace = true
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+reth-transaction-pool = { workspace = true }
+reth-primitives = { workspace = true }
+reth-chainspec = { workspace = true }
+reth-node-ethereum = { workspace = true }
```
Comment (greged93) [inline crates/pool/Cargo.toml:15]: I think you are missing a line break here

Comment (tcoratger) [reply crates/pool/Cargo.toml]: fixed

**Pull request merged**